tests/metal/simple-task.slang

```diff
@@ -0,0 +1,103 @@
+//TEST:SIMPLE(filecheck=CHECK): -entry taskMain -stage amplification -target metal
+//TEST:SIMPLE(filecheck=CHECK-ASM): -entry taskMain -stage amplification -target metallib
```

Comment:
We probably cannot run this test here because we always specify metal2.3 when we run metal compiler. We need to extend the logic to detect the use of metal 3.0 and specify the std accordingly. Before then this is going to fail.

Reply:
Should I leave this PR like this for now until this is fixed and start working on the mesh stage on a different branch, or continue working on this one?

Reply:
Let's disable this asm test by removing this line, and check this PR in.